runnable/extra-files/ldc_cabi2.cpp

```diff
@@ -0,0 +1,432 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+
+// ldc_cabi1 defines these
+extern int8_t a, b, c, d;
+extern uint32_t errors;
+
+#define TEST(b) \
+  if (!(b)) (++errors, printf("%s:%u: failed check\n", __FILE__, __LINE__))
+
+static bool testar(int8_t* a, size_t len, int8_t a0)
+{
+    for (size_t i = 0; i < len; ++i) {
+        if (a[i] != (int8_t)(a0+i)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+extern "C" {
+
+struct EMPTY {};
+struct B1 {int8_t a;};
+struct B2 {int8_t a, b;};
+struct I1 {int a;};
+struct I2 {int a, b;};
+union UI1 {int a; short b; int8_t c;};
+union UI1a {short a; int8_t b; int c;};
+struct NI1 {I1 a;};
+struct NUI1 {UI1 a;};
+union UNI1 {UI1 a; NI1 b; int c;};
+struct S3 {char a; short b;};
+struct S6 {char a; int b; char c;};
+struct S9 {char a; double b;};
+struct S19 {char a; double b, c;};
+struct F2 {float a, b;};
+struct F4 {float a, b, c, d;};
+struct D1 {double a;};
+struct D2 {double a, b;};
+struct D4 {double a, b, c, d;};
+union UD4 {D1 a; D2 b; D4 c; double e;};
+struct DA0 {double a[0];};
+struct DA4 {double a[4];};
+struct CA4 {char a[4];};
+struct DHFA1 {EMPTY a; double d;};
```

Comment:
I think they are still good tests for other architectures too.  They better pass.